test/support/utils.h

```diff
@@ -755,15 +755,15 @@ struct MaxAbsFunctor<::std::complex<_Tp>>
     }
 };
 
-struct tuple_add_fn
+struct TupleAddFunctor
 {
     template <typename Tup1, typename Tup2>
-    auto operator()(Tup1 lhs, const Tup2& rhs) const
+    auto
+    operator()(const Tup1& lhs, const Tup2& rhs) const
```

Comment:
I think in this case using a const reference is best although I may be mistaken. We don't need to forward the arguments anywhere and temporary objects can bind to const references, so I don't think a forward reference is necessary.

Is there any reason forwarding references would be preferred here?

Reply:
I think the form ```Tup1&& lhs, Tup2&& rhs`` contains less requirements about params value kind: I mean l-value, r-value and so on.

Reply:
I believe that `const Tup1& lhs` should bind to all incoming options (meaning it shouldn't put any additional requirements as compared to universal references).

I think the advantage to universal references like `Tup1&& lhs` is, as @mmichel11 mentions, is that you can keep rvalues as rvalues and forward them along as rvalues.  However, it seems that doesn't provide us anything we need in this setting.  

https://stackoverflow.com/questions/3106110/what-is-move-semantics/11540204#11540204
This is a useful table for binding from the above link:
```
            lvalue   const lvalue   rvalue   const rvalue
---------------------------------------------------------              
X&          yes
const X&    yes      yes            yes      yes
X&&                                 yes
const X&&                           yes      yes
```

I may be missing something, but I think const lvalue references are best here because universal references add complication without a clear benefit.

Reply:
OK